File: jiwook/Hash/폰켓몬.py
def solution(nums):
    answer = len(set(nums))  # 중복을 제거한 폰켓몬 종류의 개수
    if answer > len(nums) // 2:  # 만약 폰켓몬 종류 수가 선택할 수 있는 폰켓몬 수보다 많으면
        return len(nums) // 2  # N/2만큼의 폰켓몬을 선택할 수 있으므로 N/2를 반환
    return answer  # 그렇지 않으면, 폰켓몬 종류 수를 반환

File: dayeong/PGS/Hash/폰켓몬.cpp
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    unordered_map<int, int> nums_map;
    
    for(const int& n : nums){
        nums_map[n]++;
    }
    
    answer = min(nums.size()/2, nums_map.size());
    
    return answer;
}